src/OpalCompiler-Core/TemporaryVariable.class.st

```diff
@@ -16,13 +16,7 @@ TemporaryVariable >> acceptVisitor: aProgramNodeVisitor node: aNode [
 TemporaryVariable >> definingNode [
 	^ scope node temporaries 
 		detect: [ :each | each name = name ]
-		ifNone: [ 
-			" ugly workaround to support temps defined by primitives"
-			| pragma |
-			pragma := scope node methodNode pragmas detect: [ :each | each isPrimitiveError ].
-			pragma ifNil: [ ^nil ].
-			^ RBVariableNode named: (pragma argumentAt: #error:) value asString
-			]
+		ifNone: [ nil ]
```

Comment:
It would be nice to avoid nil returns. Let it fail if no node exists. And superclass method can be removed. 
There is only RBVariableNode>>isDefinition sender and the logic can be modified by #isDefinedByNode: message (returning false from Variable as default).